tests/test_component/test_task.py

```diff
@@ -422,6 +424,38 @@ def test_eq_hash(self):
         # 2nd task wasn't added since it has "uid" as the task in the set
         self.assertEqual(list(tasks_set)[0]['name'], 'unique.name.1')
 
+    # --------------------------------------------------------------------------
+    #
+    def test_annotate(self):
+
+        t1 = Task()
+        t2 = Task()
+
+        t1.annotate()
+        self.assertIsInstance(t1.annotations.inputs,     list)
+        self.assertIsInstance(t1.annotations.outputs,    list)
+        self.assertIsInstance(t1.annotations.depends_on, list)
+
+        t1.annotate(inputs=['file_1.txt', 'file_2.txt'],
```

Comment:
Based on the code, having file names with [._0-9a-z] characters should not be an issue but, more in general, maybe worth investigating if we have a module we can use to test file names with a variety of non alphanumericals? Bugs related to characters in names bitten us in the past.

Reply:
yeah, makes sense to look for corresponding libraries, but as part of other PR (I would think, we can focus on it after having similar functionality with RP)